pkg/segment/structs/evaluationstructs.go

```diff
@@ -2170,6 +2170,197 @@ func (self *NumericExpr) getSigFigs(strVal string, fltVal float64) int {
+		// according to splunks example, there is no strict checking
+		// => 123 can be treated as 123.0
+		val, err := parseFloat(valStr)
+		if err != nil {
+			return INVALID_FLOAT_DTYPE_PRINTF_ERR
+		}
+		addToBuffer(buffer, verb, val, formatValues, humanizeVal, verbCounter)
+	// works for both integer and floats
+	case 'X', 'x':
+		if hasDecimal {
+			val, err := parseFloat(valStr)
+			if err != nil {
+				return INVALID_DTYPE_PRINTF_ERR
+			}
+			addToBuffer(buffer, verb, val, formatValues, humanizeVal, verbCounter)
+		} else {
+			val, err := parseInt(valStr)
+			if err != nil {
+				return INVALID_DTYPE_PRINTF_ERR
+			}
+			addToBuffer(buffer, verb, val, formatValues, humanizeVal, verbCounter)
+		}
+	// these are verbs which don't have an equivalent in go. Therefore we handle it manually
+	case '-':
+		switch originalVerb {
+		// x, o, u - return unsigned values
+		case 'x', 'o', 'u', 'X', 'p':
+			if hasDecimal && originalVerb != 'o' {
+				if originalVerb == 'u' {
+					originalVerb = 'f'
+				} else if originalVerb == 'p' {
+					originalVerb = 'X'
+				}
+				val, err := parseFloat(valStr)
+				if err != nil {
+					return INVALID_DTYPE_PRINTF_ERR
+				}
+				if val < 0 {
+					val = val * -1
+				}
+				addToBuffer(buffer, originalVerb, val, formatValues, humanizeVal, verbCounter)
+			} else {
+				if originalVerb == 'u' {
+					originalVerb = 'd'
+				} else if originalVerb == 'o' {
+					if hasDecimal {
+						valStr = strings.Split(valStr, ".")[0]
+					}
+				} else if originalVerb == 'p' {
+					originalVerb = 'X'
+				}
+				val, err := parseInt(valStr)
+				if err != nil {
+					return INVALID_DTYPE_PRINTF_ERR
+				}
+				if val < 0 {
+					val = val * -1
+				}
+				addToBuffer(buffer, originalVerb, val, formatValues, humanizeVal, verbCounter)
+			}
+		default:
+			return INVALID_DTYPE_PRINTF_ERR
+		}
+	// rest are only string formatters which require a string argument
+	default:
+		addToBuffer(buffer, verb, valStr, formatValues, humanizeVal, verbCounter)
+	}
+	return nil
+}
+
+func splPrintfHandleArgs(verbCounter int, fieldToValue map[string]sutils.CValueEnclosure, self *TextExpr) (string, error) {
+	var valStr string
+	var err error
+	if verbCounter < len(self.ValueList) {
+		fieldName := self.ValueList[verbCounter].FieldName
+		if fieldName != "" {
+			fltVal, err := handleNoArgFunction(fieldName)
+			if err != nil {
+				if val, ok := fieldToValue[fieldName]; ok {
+					switch val.Dtype {
+					case sutils.SS_DT_STRING:
+						valStr = val.CVal.(string)
+					case sutils.SS_DT_SIGNED_NUM:
+						valStr = strconv.FormatInt(val.CVal.(int64), 10)
+					case sutils.SS_DT_UNSIGNED_NUM:
+						valStr = strconv.FormatUint(val.CVal.(uint64), 10)
+					case sutils.SS_DT_FLOAT:
+						valStr = strconv.FormatFloat(val.CVal.(float64), 'f', -1, 64)
+					default:
+						return "", fmt.Errorf("TextExpr.Evaluate: printf-splunkPrintfHandleArgs: Unable to determine dtype for value in fieldToValueMap; val: %v", val)
```

Comment:
```suggestion
						return "", fmt.Errorf("splPrintfHandleArgs: Unable to determine dtype for value in fieldToValueMap; val: %v", val)
```

similar comment elsewhere